### PR TITLE
[FIX] restore_checkpoint without mesh group

### PR DESCRIPTION
**Pull request by zw123han**
`alpa.restore_checkpoint` only works when a `PhysicalDeviceMeshGroup` (eg. `VirtualPhysicalMesh`) exists in cases such as `PipeshardParallel`. 
However, `ShardParallel` uses `PhysicalDeviceMesh`, meaning `get_global_virtual_physical_mesh().launched_physical_mesh_group` will return None. 

This PR bypasses the mesh_group in these cases to allow `ShardParallel` checkpoints to be restored to the physical mesh.

### CODE REVIEW

File: alpa/serialization.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 import numpy as np
 
 from alpa.device_mesh import (DistributedArray, ReplicatedDistributedArray,
-                              get_global_virtual_physical_mesh)
+                              get_global_virtual_physical_mesh, get_global_physical_mesh)
```
Comment (merrymercy) [inline alpa/serialization.py:18]:
```suggestion
                              get_global_virtual_physical_mesh,
                              get_global_physical_mesh)
```

**Comment by merrymercy** [comment]
Please fix the lint

File: alpa/serialization.py [hunk 2]
```diff
@@ -15,7 +15,8 @@
 import numpy as np
 
 from alpa.device_mesh import (DistributedArray, ReplicatedDistributedArray,
-                              get_global_virtual_physical_mesh)
+                              get_global_virtual_physical_mesh,
```
Comment (merrymercy) [inline alpa/serialization.py:18]:
```suggestion
                              get_global_virtual_physical_mesh,
```

File: alpa/serialization.py [hunk 3]
```diff
@@ -156,13 +157,21 @@ def restore_checkpoint(ckpt_dir: Union[str, os.PathLike], step: int,
...
+    assert mesh_group is not None or physical_mesh is not None
 
     for path, info in zip(state_paths, flat_info):
         if info is None:
             logger.warning("Variable is not used, skip loading it")
             flat_load_state.append(None)
-        if len(info.mesh_ids) == 1:
+        elif mesh_group is None:
+            dist_arr = DistributedArray.load(os.path.join(ckpt_dir,
+                                                          path), info.aval,
```
Comment (merrymercy) [inline alpa/serialization.py:170]: dist_arr = DistributedArray.load(os.path.join(ckpt_dir, path),
                                             info.aval, physical_mesh,

Comment (merrymercy) [inline alpa/serialization.py:170]:
```suggestion
            dist_arr = DistributedArray.load(os.path.join(ckpt_dir, path),
                                             info.aval, physical_mesh,
```

**Comment by zw123han** [comment]
Sorry pressed the wrong button, please cancel that last merge request. 
Will fix yapf, physical_mesh arg was duplicated.

**Pull request merged**